server/src/main/java/io/pravega/schemaregistry/service/Main.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.service;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.schemaregistry.server.rest.RestServer;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import io.pravega.schemaregistry.storage.SchemaStore;
+import io.pravega.schemaregistry.storage.SchemaStoreFactory;
+import io.pravega.schemaregistry.storage.StoreType;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URI;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Slf4j
+public class Main {
+    public static void main(String[] args) {
+        ClientConfig clientConfig = ClientConfig.builder().controllerURI(URI.create(Config.PRAVEGA_CONTROLLER_URI)).build();
```

Comment:
If Auth and TLS are enabled, there will be additional params needed to be set in the `ClientConfig`.

Reply:
auth and tls are enabled for schema registry service.

for talking to controller --> 
tls: from controller url
auth: the creds are extracted in ClientConfig.builder automatically from the system properties.

Reply:
W.r.t. TLS: 
If TLS is enabled for Controller, the Controller URI will have `tls:` as the scheme. 

In most cases one'd access both the Controller and Segment store over TLS. The `tls` scheme tells the client to access both over TLS. 

In some cases, the Controller service has TLS disabled, but Segment store has it enabled. That is done when TLS terminates at the reverse proxy (or say, Kubernetes Ingress). 
* If the controller URI has the address of the reverse proxy, setting the scheme `tls` would work just fine.
* However, if the Controller service is hit directly, the URI will have `tcp` as the Controller has TLS disabled. In this case, you also have to set the flag to enable segment store TLS explicitly. Doing so enables the client to talk to Controller over plaintext and to Segment Store over TLS. 

In general, users are going to use a Controller URI with scheme `tls` n both cases: 

* TLS terminates at the services for both Controller and Segment Store
* TLS terminates at the service for Segment Store and at the front-ending reverse-proxy/ingress for the Controllers. (just that the address will be the Ingress/proxy's rather than the Controller service, in this case.)

So, we can make an assumption that setting the controller scheme as `tls` is enough. 

We also need to set the truststore. If the server uses a custom CA (which is very common for internal applications) to sign the certificates used in Controller/Segment Store/proxy, the CA's cert needs to be supplied so that the client can trust the server's cert. 

In addition, there is another param we need to worry about - the validateHostName. That too should be taken as input from the admin. 

W.r.t. Auth:

- You might want to create custom config names for schema registry as opposed to using the client system properties or environment variables to avoid leaking client internals to the admins.